csrc/ir/builder.cpp

```diff
@@ -340,19 +340,33 @@ Val* SimplifyingIrBuilder::addExpr(
     rhs_dtype = getDataType(rhs);
   }
   if (lhs == nullptr) {
-    return IrBuilder::IrBuilder::create<Val>(rhs, rhs_dtype);
+    return IrBuilder::create<Val>(rhs, rhs_dtype);
+  }
+  // simplify x + y - y as x
```

Comment:
Curious why this specific pattern?

Reply:
In the parallel dim map, due to warp specialization, we will generate Vals like `x + 1`, where `x` is the original parallel dimension as inferred, and `1` is the extra one for warp specialization. But sometimes, we need to get the original one, so we will generate things like `x + 1 - 1`. It is important to simplify this as `x`, because, in some environment, we my not have a fusion guard (for example, `codegen.cpp` is not supposed to modify the kernel, so I can not create an IR node `x + 1 - 1`, and I need to use `SimplifyingIrBuilder` to find the pointer for `x`).